### DISCUSSION TITLE
Can we set the properties of activities from database.

### DISCUSSION THREAD

**Discussion by kashish237** [elsa-workflows/elsa-core#525] [General]
Presently we are writing the code to set the properties for activities.
Can we set the restriction that a particular activity cannot be connected to  another particular activity.

This make it bit hardcoded.

**Comment by sfmskywalker** [comment]
Short answer: not as a generic built-in rule in the current Studio/Core stack.

In the current designer, connection validity is enforced at the port level. `AddWorkflowsModule()` registers the default activity port providers and the workflows designer (`elsa-studio/src/modules/Elsa.Studio.Workflows/Extensions/ServiceCollectionExtensions.cs:33-56`). Those port providers determine which ports an activity exposes; for example, `DynamicOutcomesPortProvider` generates flow ports from activity metadata and configured outcomes (`elsa-studio/src/modules/Elsa.Studio.ActivityPortProviders/Extensions/ServiceCollectionExtensions.cs:15-24`, `elsa-studio/src/modules/Elsa.Studio.ActivityPortProviders/Providers/DynamicOutcomesPortProvider.cs:17-53`).

The React Flow designer's current connection check then validates only that the source handle is an `out` port and the target handle is an `in` port (`elsa-studio/src/modules/Elsa.Studio.Workflows.Designer/ClientLib/src/react-designer/components/Designer.tsx:947-960`).

So if your requirement is "activity A must never connect to activity B", there isn't a general-purpose built-in rule for arbitrary activity-pair restrictions at that level today. The supported way to constrain authoring is to shape the ports/outcomes the activity exposes, or to add custom designer/validation logic on top.

Closing this as answered, but if you want to pursue a specific restriction pattern, opening a new discussion with the exact activity types and desired rule would make that easier to evaluate.
